internal/controllers/topology/cluster/patches/inline/json_patch_generator_test.go

```diff
@@ -876,6 +1184,39 @@ func TestMatchesSelector(t *testing.T) {
 			},
 			match: true,
 		},
+		{
+			name: "Match MP InfrastructureMachineTemplate",
+			req: &runtimehooksv1.GeneratePatchesRequestItem{
+				Object: runtime.RawExtension{
+					Object: &unstructured.Unstructured{
+						Object: map[string]interface{}{
+							"apiVersion": "infrastructure.cluster.x-k8s.io/v1beta1",
+							"kind":       "AzureMachineTemplate",
+						},
+					},
+				},
+				HolderReference: runtimehooksv1.HolderReference{
+					APIVersion: clusterv1.GroupVersion.String(),
+					Kind:       "MachinePool",
+					Name:       "my-mp-0",
+					Namespace:  "default",
+					FieldPath:  "spec.template.spec.infrastructureRef",
+				},
+			},
+			templateVariables: map[string]apiextensionsv1.JSON{
+				"builtin": {Raw: []byte(`{"machinePool":{"class":"classA"}}`)},
+			},
+			selector: clusterv1.PatchSelector{
+				APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
+				Kind:       "AzureMachineTemplate",
```

Comment:
Shouldn't that be AzureMachinePool in the case of a MachinePool?

Reply:
Nice catch!